### DISCUSSION TITLE
Print variable set in $GITHUB_ENV

### DISCUSSION THREAD

**Discussion by reikdas** [community/community#26392] [Actions]
<p>How do I get the value of an environment variable set in <code>$GITHUB_ENV</code>?</p>
<p>For eg -<br>
<code>echo "CC=clang" &gt;&gt; $GITHUB_ENV</code></p>
<p>Now I want to do -<br>
<code>$CC --version</code><br>
but <code>$CC</code> does not exist.</p>

**Comment by airtower-luna** [comment]
<p>Could you post the context from your workflow file? That should work, assuming the second command is in another <code>run</code> step in the same job.</p>

**Comment by Simran-B** [comment]
<p>You might want to set it as output so that you can explicitly reference and set it as env instead. But would need to see your workflow to better understand the problem. Do you use PowerShell by any chance?</p>

**Comment by reikdas** [comment]
<p>This is my config file - <a href="https://github.com/reikdas/cling-1/blob/090f490dae44e7f4857ac9310d2c01724f9ef3aa/.github/workflows/ci.yml" rel="noopener nofollow ugc">https://github.com/reikdas/cling-1/blob/090f490dae44e7f4857ac9310d2c01724f9ef3aa/.github/workflows/ci.yml</a> (Note where I try <code>$CC --version</code>)</p>
<aside class="quote no-group" data-username="Simran-B" data-post="3" data-topic="141006">
<div class="title">
<div class="quote-controls"></div>
 Simran-B:</div>
<blockquote>
<p>Do you use PowerShell by any chance?</p>
</blockquote>
</aside>
<p>No. Bash.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I see the problem there: The <code>$CC --version</code> is part of the same steps the set write <code>CC</code> to <code>GITHUB_ENV</code>. That’s not going to work, you can only use it in later steps. So, remove that command from the “setup compiler” steps, and add one like this after them:</p>
<pre><code class="lang-yaml">    - name: Check compiler version
      run: '$CC --version'
</code></pre>

**Comment by Simran-B** [comment]
<p>If you only need an environment variable within the current step, then there is no need to involve GITHUB_ENV at all. But I guess some of the commands in later steps take CC and CXX into account?</p>
<p>To make <code>$CC --version</code> work using a single step, you could also set it as regular variable for immediate access:</p>
<pre><code class="lang-yaml">   - name: Setup compiler on Linux
      run: |
        sudo apt-get update
        if [ "${{ matrix.compiler }}" = "gcc" ]; then
          echo "CC=gcc" &gt;&gt; $GITHUB_ENV
          echo "CXX=g++" &gt;&gt; $GITHUB_ENV
          CC=gcc
        else
          sudo apt-get install -y clang
          echo "CC=clang" &gt;&gt; $GITHUB_ENV
          echo "CXX=clang++" &gt;&gt; $GITHUB_ENV
          CC=clang
        fi
        $CC --version
      if: runner.os == 'Linux'
      shell: bash
</code></pre>

**Comment by jetersen** [comment]
<p>Here is how to do it in powershell 5.1 or newer</p>
<p><a class="mention" href="/simran-b">@Simran-B</a></p>
<pre><code class="lang-yaml">      - name: Increment version
        shell: pwsh
        run: |
          $version = [version]((git describe --tags --abbrev=0) -replace 'v','')
          $increment = [version]::New($version.Major,$version.Minor,$version.Build+1)
          "REVISION=$increment" | Add-Content -Path $ENV:GITHUB_ENV
</code></pre>

**Comment by Simran-B** [comment]
<p>I wouldn’t recommend <code>"REVISION=$increment" | Add-Content -Path $ENV:GITHUB_ENV</code> in PowerShell (Core) below version 6, because of the inconsistent handling of character encoding (and sometimes addition of a byte-order mark):</p>
<ul>
<li><a href="https://stackoverflow.com/a/34969243/2044940">StackOverflow: Using PowerShell to write a file in UTF-8 without the BOM</a></li>
<li><a href="https://stackoverflow.com/a/40098904/2044940">StackOverflow: Changing PowerShell’s default output encoding to UTF-8</a></li>
</ul>
<p>I suppose in PowerShell Core (<code>pwsh</code>) as available in GitHub hosted runners (version 7.x) it’s fine, because it defaults to BOM-less UTF-8 for everything as far as I know. That would mean this should work just fine:</p>
<pre><code>echo "foo" &gt;&gt; $Env:GITHUB_ENV
</code></pre>
<p>Nonetheless, the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files">official recommendation</a> is:</p>
<pre><code>echo "foo" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
</code></pre>
<p>It doesn’t say whether this is for the legacy PowerShell only, but it might.</p>
